engineering/backend/restful.md

```diff
@@ -33,27 +33,65 @@ Always version your API. Version via the URL, not via headers. Versioning APIs a
 ## Filter, Search and Sort
 Use query parameters for advanced filtering, sorting & searching.
 
-Pagination:
+###Pagination
+
+**Offset pagination:**
+
+- `page` - current page wanted to fetch
+- `size` - number of items per page
+
+- The API should return `totalItems` and `totalPages`
+
+Example:
 ```
 GET /companies?page=23&size=50
 ```
 
-Sort:
+**Cursor pagination**
+
+- `cursor` - is the reference element in the list
+- `size` - number of items per page
+
+- The API should return `previousCursor` and `nextCursor`
```

Comment:
should describe how we structure the cursor from the backend and use from client